spark_log_parser/__main__.py

```diff
@@ -12,17 +11,25 @@
 
 logger = logging.getLogger("spark_log_parser")
 
+
 parser = argparse.ArgumentParser("spark_log_parser")
-parser.add_argument("-l", "--log-file", required=True, type=Path, help="path to event log")
 parser.add_argument(
-    "-r", "--result-dir", required=True, help="path to directory in which to save parsed logs"
+    "-l", "--log-file", required=True, type=Path, help="path to event log file or directory"
+)
+parser.add_argument(
+    "-r",
+    "--result-dir",
+    required=True,
+    type=Path,
+    help="path to directory in which to save the parsed log",
 )
 args = parser.parse_args()
 
-if not os.path.isdir(args.result_dir):
+if not args.result_dir.is_dir():
     logger.error("%s is not a directory", args.result_dir)
     sys.exit(1)
```

Comment:
I think we should refrain from using `sys.exit()` here. In the case the spark_log_parser is used anywhere in the backend or celery task, it would terminate the process, and likely cause unintended side effects.

Reply:
I think we're ok here. This code is only executed by users on the command line, e.g.
```
python -m spark_log_parser
```

Reply:
ok, makes sense. It caught my eye because some areas we raise a `ValueError` but here we exit the process.